src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -364,6 +357,27 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return true;
     }
 
+    // Method Description:
+    // - Setup our event handlers for this connection. If we've currently got a
+    //   connection, then this'll revoke the existing connection's handlers.
+    void ControlCore::_setConnection(const TerminalConnection::ITerminalConnection& connection)
+    {
+        if (_connection)
+        {
+            _connectionOutputEventRevoker.revoke();

```

Comment:
overwriting `_connectionStateChangedRevoker` later _should_ revoke it automatically, without you needing to check this